### PR TITLE
Move UIState instance to app.js

### PR DESCRIPTION
**Pull request by hatched**
The react components need access to the UIState instance to know how to properly update when the state changes.

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

Play around with the GUI to make sure it works as expected.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2831/
Test PASSed.

File: jujugui/static/gui/src/app/app.js [hunk 1]
```diff
@@ -684,6 +688,23 @@ YUI.add('juju-gui', function(Y) {
...
+      @param {Boolean} sandbox
+      @param {String} baseUrl
+    */
+    _setupUIState: function(sandbox, baseUrl) {
+      this.state = new models.UIState({
+        // Disallow routing to inspectors if we are in sandbox mode; the
+        // model to be inspected will not be available.
+        allowInspector: !sandbox,
+        baseUrl: baseUrl || '',
+        dispatchers: {}
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/app.js]: Maybe a comment here to why we don't set any initial dispatchers would be helpful.

File: jujugui/static/gui/src/app/subapps/browser/browser.js [hunk 2]
```diff
@@ -175,30 +175,26 @@ YUI.add('subapp-browser', function(Y) {
...
-            machine: this._machine.bind(this),
-            empty: this.emptySectionB.bind(this)
-          }
-        }
-      });
 
       this._registerSubappHelpers();
 
+      this.state = cfg.state
+      this.state.set('dispatchers', {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/subapps/browser/browser.js]: Maybe we could set this up as a merge now so we don't have to in the future. At the least maybe a comment that this needs to be updated to merge.

**Comment by huwshimi** [comment]
Nice one @hatched! A couple of tiny comments :+1:.

File: jujugui/static/gui/src/app/app.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/app.js]: done

File: jujugui/static/gui/src/app/subapps/browser/browser.js (hunk 2 above)
Comment (hatched) [inline jujugui/static/gui/src/app/subapps/browser/browser.js]: done

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/2833/
Test PASSed.

**Comment by anthonydillon** [comment]
Code looks good :+1:. No QA

**Comment by hatched** [comment]
Thanks for the reviews! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**